### PR TITLE
UP-4875:  Provide Gradle-based replacement for 'ant initdb'

### PR DESCRIPTION
**Pull request by drewwills**
https://issues.jasig.org/browse/UP-4875

##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] documentation is changed or added

##### Description of change

This PR defines & implements the `dataInit` Gradle task, which does (nearly) the same things `$ ant initdb` did in the Ant/Maven build.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by bpowell** [comment]
I like the name change of `uportal-data` to `data`. I'd prefer for `base` to be renamed to `default` and a new folder called something like `overrides`. Where you generally don't touch `default`, all customization goes into the `overrides` folder. That way upgrades are much easier for us to handle. Doing this we will know exactly what we SHOULD be messing with.

**Comment by drewwills** [comment]
Where you generally don't touch default, all customization goes into the overrides folder. That way upgrades are much easier for us to handle. Doing this we will know exactly what we SHOULD be messing with.

I think we're 100% on the same page as far as this notion.

The reason I like `base` and `{your_name_here}` is that -- with few exceptions -- there's no overlap between the files in one directory vs. the other.

The `base` data set is...
  - (1) The minimal  set of stuff you need for a functioning uPortal instance
  - (2) Behind-the-scenes stuff (e.g. permission-owners and admin portlets) that you shouldn't customize

The other folder is your _Implementation Data Set_.  We'll provide one -- `quickstart` -- but you're not intended to customize that either.  You're supposed to clone it and give it a local name (e.g. `mySail`), then go into build.properties and specify...

```
implementation.entities.location=data/mySail
```

**Comment by drewwills** [comment]
@bpowell, @ChristianMurphy, et al. -- this PR is ready for merge.  I'm targeting Monday morning for that event.

I'm still open-minded about names.  We can continue the process of choosing & discussing them for the next several weeks, up to the point where we choose to release uP5.

**Comment by bpowell** [comment]
@drewwills if you are going to have a `quickstart` directory, maybe add a REAMDE.md file here to say rename `quickstart` to your own name and add your files here. To me `quickstart` seemed a little odd. I wasn't sure when I first started on uPortal if I should just delete it or what happens when I do. That's why I thought something like `overrides` or even `overlay` would be better named. Overall this is a great step forward.

File: README.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ recommended for server deployments)
 
 - **dataInit**:  Drops and recreates the schema in the embedded HSQLDB uPortal database, then loads
 both the Base Data Set and the Implementation Data Set
+- **dataImport**:  Imports the specified entity file(s) into the embedded HSQLDB uPortal database;
```
Comment (ChristianMurphy) [inline README.md]: could the CLI arguments be labeled as code snippets? `` `-Dfile={path}` ``

File: README.md [hunk 2]
```diff
@@ -2,15 +2,45 @@
...
 
-### Gradle Tasks
+### Using Gradle
+
+`uPortal-start` provides a build system and several CLI tools through Gradle, and it comes with a
+_Gradle Wrapper_ so you don't have to install Gradle to use it.
+
+Invoking the Gradle Wrapper on *nix:
+
+```
```
Comment (ChristianMurphy) [inline README.md]: Could this be flagged as a shell script?

File: README.md [hunk 3]
```diff
@@ -2,15 +2,45 @@
...
+    $ ./gradlew {taskname} [{taskname}...]
+```
+
+Invoking the Gradle Wrapper on Windows:
+
+```
```
Comment (ChristianMurphy) [inline README.md]: could this be flagged as a batch script?

File: README.md [hunk 4]
```diff
@@ -2,15 +2,45 @@
...
-### Gradle Tasks
+### Using Gradle
+
+`uPortal-start` provides a build system and several CLI tools through Gradle, and it comes with a
+_Gradle Wrapper_ so you don't have to install Gradle to use it.
+
+Invoking the Gradle Wrapper on *nix:
+
+```
+    $ ./gradlew {taskname} [{taskname}...]
```
Comment (ChristianMurphy) [inline README.md]: :art: Including a leading `$` can reduce copy-ability.

File: README.md [hunk 5]
```diff
@@ -2,15 +2,45 @@
...
+```
+
+Invoking the Gradle Wrapper on Windows:
+
+```
+    > gradlew.bat {taskname} [{taskname}...]
```
Comment (ChristianMurphy) [inline README.md]: :art: including a leading `>` can reduce copy-ability

**Pull request merged**